### PR TITLE
Defer sendmessage

### LINKED ISSUE #154: cross movie messages sometimes fail
Here is a simple reduction of the bug:

http://jsfiddle.net/vnFLH/1/

Source code:
https://gist.github.com/4123564

Note that the `on load` callback function is always triggered but the message doesn't always get sent (which is the bug).

Tested in Chrome 23.0.1271.64 and Safari 6.0.2 (8536.26.17) on OS X 10.8.2

### PR DESCRIPTION
**Pull request by davidaurelio**
This defers sending of user messages to the runner context through `sendMessage()` until the runner has started up and is able to receive them.

Would fix #154

### CODE REVIEW

**Comment by iamdustan** [comment]
I am pretty sure this also fixes my issue in #100 and potentially #150?

**Comment by davidaurelio** [comment]
If the `'load'` event is acceptable, this fixes #100, will check #150 when it is tomorrow at my place :)

**Comment by iamdustan** [comment]
I’ll pull this branch in tomorrow and remove the code I’ve added to work around it. If it does the docs https://github.com/uxebu/bonsai-docs/blob/gh-pages/overview/Communication.md here (I think) will be able to be updated. I’ll verify tomorrow.

**Comment by iamdustan** [comment]
:thumbsup: Just tested it and it is working like a charm.

File: src/renderer/renderer_controller.js [hunk 1]
```diff
@@ -310,6 +327,14 @@ function(tools, EventEmitter, URI) {
      * @returns {this} The instance
      */
     sendMessage: function(category, messageData) {
+      var pendingMessages = this._pendingMessages;
+      if (pendingMessages) {
```
Comment (klipstein) [inline src/renderer/renderer_controller.js:341]: could you be more explicit here? it probably is sufficient to add a comment which describes that it is set to null when the runner is ready.

Comment (davidaurelio) [inline src/renderer/renderer_controller.js:341]: of course

**Comment by mattetti** [comment]
Very nice, thank you very much!

**Pull request merged**